src/carve/api.clj

```diff
@@ -0,0 +1,21 @@
+(ns carve.api
+  (:refer-clojure :exclude [run!])
+  (:require [carve.impl :as impl]))
+
+(defn print! [{:keys [:report :config]}]
+  (let [format (-> config :report :format)]
+    (impl/print-report report format)))
+
+(defn run!
+  "Returns unused vars as EDN data in :report. The entire return value
+  may be passed to print! for printing. Accepts similar options as the
+  CLI."
+  ([] (run! nil))
+  ([opts]
+   (impl/run+ opts)))
+
+(defn report
+  "Similar to run! but forces non-side-effecting non-interactive behavior."
+  ([] (report {:merge-config true}))
+  ([opts]
+   (impl/run+ (assoc opts :report true))))
```

Comment:
WDYT about passing `:silent true` as well? Currently it doesn't make a difference, but that seems subject to change e.g. you might add logging in a new place tomorrow.

Reply:
`:silent` is about writing stuff to stdout for interactivity. You can rely on `:report` always implying `:silent true`.

Logging warnings (like in the main) may happen to warn users of important stuff like breakage or something, despite this setting. In that case you can rebind `*err*` to a new StringWriter to ignore the output, if that should happen.